gspread/utils.py

```diff
@@ -519,19 +549,23 @@ def extract_id_from_url(url):
     raise NoValidUrlKeyFound
 
 
-def wid_to_gid(wid):
+def wid_to_gid(wid: str) -> str:
     """Calculate gid of a worksheet from its wid."""
     widval = wid[1:] if len(wid) > 3 else wid
     xorval = 474 if len(wid) > 3 else 31578
     return str(int(widval, 36) ^ xorval)
 
 
-def rightpad(row, max_len):
+def rightpad(row: List[Any], max_len: int) -> List[Any]:
     pad_len = max_len - len(row)
     return row + ([""] * pad_len) if pad_len != 0 else row
 
 
-def fill_gaps(L, rows=None, cols=None):
+def fill_gaps(
+    L: List[Any],
+    rows: Optional[int] = None,
+    cols: Optional[int] = None
+) -> List[Any]:
```

Comment:
looking at the code I see that it should return too `List[List[Any]]` and we should update the return value in case of value error to: `[[]]` please.